procedures.sql

```diff
@@ -0,0 +1,1136 @@
+
+CREATE PROCEDURE UpdateMeeting
+    @meeting_id INT,
+    @module_id INT = NULL,
+    @subject_id INT = NULL,
+    @date DATETIME = NULL,
+    @student_limit INT = NULL,
+    @type VARCHAR(10) = NULL,
+    @url VARCHAR(200) = NULL,
+    @substituting_room_id INT = NULL,
+    @substituting_teacher_id INT = NULL,
+    @translator_id INT = NULL,
+    @standalone_price INT = NULL
+AS
+BEGIN
+    IF @module_id IS NOT NULL OR @subject_id IS NOT NULL
+        BEGIN
+            UPDATE Meeting
+            SET module_id = @module_id,
+                subject_id = @subject_id
+            WHERE id = @meeting_id;
+        END
+    IF @date IS NOT NULL
+        BEGIN
+            UPDATE Meeting
+            SET date = @date
+            WHERE id = @meeting_id;
+        END
+    IF @student_limit IS NOT NULL
+        BEGIN
+            UPDATE Meeting
+            SET student_limit = @student_limit
+            WHERE id = @meeting_id;
+        END
+    IF @type IN ('online, video')
+        BEGIN
+            UPDATE Meeting
+            SET type = @type,
+                url = NULL
+            WHERE id = @meeting_id;
+        END
+    IF @type LIKE 'in_person'
+        BEGIN
+            UPDATE Meeting
+            SET type = @type
+            WHERE id = @meeting_id;
+        END
+    IF @url IS NOT NULL
+        BEGIN
+            UPDATE Meeting
+            SET url = @url
+            WHERE id = @meeting_id;
+        END
+    IF @substituting_room_id IS NOT NULL
+        BEGIN
+            UPDATE Meeting
+            SET substituting_room_id = @substituting_room_id
+            WHERE id = @meeting_id;
+        END
+    IF @substituting_teacher_id IS NOT NULL
+        BEGIN
+            UPDATE Meeting
+            SET substituting_teacher_id = @substituting_teacher_id
+            WHERE id = @meeting_id;
+        END
+    IF @translator_id IS NOT NULL
+        BEGIN
+            UPDATE Meeting
+            SET translator_id = @translator_id
+            WHERE id = @meeting_id;
+        END
+    IF @standalone_price IS NOT NULL
+        BEGIN
+            UPDATE Meeting
+            SET standalone_price = @standalone_price
+            WHERE id = @meeting_id;
+        END
+END;
+GO
+-- EXEC UpdateMeeting 1, @substituting_teacher_id = 27, @substituting_room_id = 17
+
+
+
+CREATE PROCEDURE AddStudentToMeeting
+    @student_id INT,
+    @meeting_id INT,
+    @payment_date DATETIME = NULL
+AS
+BEGIN
+    INSERT INTO StudentMeeting (student_id, meeting_id, payment_date)
+    VALUES (@student_id, @meeting_id, @payment_date);
+END;
+GO
+-- EXEC AddStudentToMeeting 17, 23, '2021-01-01 12:00:00'
+
+CREATE PROCEDURE RemoveStudentFromMeeting
+    @student_id INT,
+    @meeting_id INT
+AS
+BEGIN
+    DELETE FROM StudentMeeting
+    WHERE student_id = @student_id AND meeting_id = @meeting_id;
+END;
+GO
+-- EXEC RemoveStudentFromMeeting 17, 23
+
+CREATE PROCEDURE PayForMeeting
```

Comment:
you don't pay for the meeting but u register the payment

Reply:
ok